torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,326 @@
+import collections
+import dataclasses
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+from typing import List, Tuple
+
+import torch
+
+
+@dataclasses.dataclass(repr=False, eq=False, frozen=True)
+class CallgrindStats(object):
+    stmt: str
+    setup: str
+    number: int
+    num_threads: int
+    built_with_debug_symbols: bool
+    baseline_inclusive_stats: List[Tuple[int, str]]
+    baseline_exclusive_stats: List[Tuple[int, str]]
+    stmt_inclusive_stats: List[Tuple[int, str]]
+    stmt_exclusive_stats: List[Tuple[int, str]]
+
+    def __repr__(self):
+        instruction_count = str(sum(c for c, _ in self.stmt_exclusive_stats))
+        baseline_count = str(sum(c for c, _ in self.baseline_exclusive_stats))
+        count_len = max(len(instruction_count), len(baseline_count))
+
+        # Pad lines after the first to align properly.
+        stmt = self.stmt.replace('\n', '\n' + ' ' * 9)
+        setup = self.setup.replace('\n', '\n' + ' ' * 9)
+        lines = [
+            f"{super().__repr__()}",
+            f"  stmt:  {stmt}",
+            f"  setup: {setup}",
+            f"  {self.num_threads} thread{'s' if self.num_threads > 1 else ''}",
+            f"  Instructions: {instruction_count.rjust(count_len)}",
+            f"  Baseline:     {baseline_count.rjust(count_len)}",
+        ]
+        if not self.built_with_debug_symbols:
+            lines.extend([
+                "Warning: PyTorch was not built with debug symbols.",
+                "         Source information may be limited. Rebuild with",
+                "         REL_WITH_DEB_INFO=1 for more detailed results.",
+            ])
+        return "\n".join(lines)
+
+    def stats(self, inclusive=False):
+        if inclusive:
+            first, second = self.stmt_inclusive_stats, self.baseline_inclusive_stats
+        else:
+            first, second = self.stmt_exclusive_stats, self.baseline_exclusive_stats
+        return self._diff(first, second)
+
+    def as_standardized(self):
+        def strip_prefix(stats):
+            counts = collections.defaultdict(int)
+
+            # "Python" and "Objects" come from CPython.
+            prefix_truncations = ("build/aten/", "Python/", "Objects/")
```

Comment:
So.... we better ever have a directory in our repo that is named `Python` or `Objects`?

Reply:
I assume you meant "never"? This is just an (optional, default off) rule for pruning out some of the long prefixes. Unless we also have the same filename and method, it wouldn't collide with the CPython name. The issue that this is trying to address is that if you diff callgrind counts and the prefixes differ, it drowns out the real delta with a pair of large positive and negative entries.

Reply:
What you wrote here, should go in the doblock for `as_standardized`

Reply:
Done.